iep-010/README.adoc

```diff
@@ -0,0 +1,81 @@
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+= IEP-010: Update Center on Azure
+
+:toc:
+
+.Metadata
+[cols="2"]
+|===
+| IEP
+| 10
+
+| Title
+| Update Center on Azure
+
+| Author
+| link:https://github.com/olblak[Olivier Vernin]
+
+| Status
+| :speech_balloon: In-process
+
+| Type
+| [Architecture]
+
+| Created
+| 2018-06-01
+|===
+
+== Abstract
+The link:https://updates.jenkins.io[update center] is a web service that provides information to jenkins instances in order for them to know if core/plugins need to be updated and where to find appropriated updates.
+This service is systematically queried by each jenkins instances. It's currently running on the AWS account and need to be migrated to Azure.
+This service is a key element of the way we distribute jenkins packages.
+
+== Specification
+=== Current State
+Currently the service is running on an EC2 instance and it's composed of the following elements:
+
+* An apache webserver that listen on two endpoints.
+** link:http://updates.jenkins-ci.org[updates.jenkins-ci.org]
+** link:http://updates.jenkins.io[updates.jenkins.io]
+* A data directory populated by a shell script ``link:https://github.com/jenkins-infra/update-center2/blob/master/site/generate.sh[generate.sh]`` which is run every 4 minutes from trusted.ci. This script uploads datas via rsync on a linux a machine.
+
+Although the current architecture is quiete stable, there are still places for improvements.
+
+. This web service only contains static files and it would more reliable to just store them on an Azure blob storage.
```

Comment:
It is not clear to me that this is more reliable. Plus even a small reliability benefit seems offset by tying us to a cloud vendor even more.

Wasn't this the service that caused a several day outage just recently? And this IEP proposes to migrate more stuff there?

Reply:
Without being the priority atm, the machine running the update-center needs to be work on. It's a quite big (expensive) machine running on AWS (64GB of Ram, 16VCPU) with way too much responsibility. 
This IEP document would need to be changed anyway, many things changed since Jun 2018

Reply:
It's safer to remove one service at the time from that machine, we already offload the mirror service

Reply:
Right, I'm not saying we shouldn't clean up our confusing infra, I'm just not sure about the assertion that Azure BS is better than serving static rsynced files from a simple web server.